packages/datadog-instrumentations/src/fs.js

```diff
@@ -19,11 +19,18 @@ const twoPathMethods = ['copyFile', 'link', 'rename', 'symlink']
 const twoPathMethodsSync = ['copyFileSync', 'linkSync', 'renameSync', 'symlinkSync']
 
 addHook({ name: 'fs' }, fs => {
-  shimmer.massWrap(fs, onePathMethods.concat(onePathMethodsSync), wrapFsMethod(fsChannel, 1))
-  shimmer.massWrap(fs, twoPathMethods.concat(twoPathMethodsSync), wrapFsMethod(fsChannel, 2))
+  const allOnePathMethods = onePathMethods.concat(onePathMethodsSync)
+  const allTwoPathMethods = twoPathMethods.concat(twoPathMethodsSync)
+  const allFsProperties = Object.keys(fs)
+  const allFsPromisesProperties = Object.keys(fs.promises)
+  shimmer.massWrap(fs, allFsProperties.filter(name => allOnePathMethods.includes(name)), wrapFsMethod(fsChannel, 1))
+  shimmer.massWrap(fs, allFsProperties.filter(name => allTwoPathMethods.includes(name)), wrapFsMethod(fsChannel, 2))
+
+  shimmer.massWrap(fs.promises, allFsPromisesProperties.filter(name => onePathMethods.includes(name)),
+    wrapFsMethod(fsChannel, 1))
+  shimmer.massWrap(fs.promises, allFsPromisesProperties.filter(name => twoPathMethods.includes(name)),
+    wrapFsMethod(fsChannel, 2))
```

Comment:
Seems to me like it would make more sense to just make another methods array for promises rather than joining everything and then filtering it back again. 🤔

Reply:
Yes, I can create another list with promise methods. 
But I think that filtering is a good idea, to be safe if nodejs decides to deprecate and delete any method.

Reply:
I doubt any of those methods will go away any time soon. If they get deprecated at all we'll have _years_ of warning. 😅

Reply:
I'm going to add the filter again, it look like there are methods in node 14 that the are not in node 12, for example, `rm` method.